README.md

```diff
@@ -98,5 +104,68 @@ let expected = Data {
 assert_eq!(output, expected);
 ``` 
 
+### Bucket
+
+Since the above idiom (a subspace for a class of items) is so common and useful, 
+and there is no easy way to return this from a function 
+(bucket holds a reference to space, and cannot live longer than the local variable), the two are often
+combined into a `Bucket`. A Bucket works just like the example above, except the creation can be
+in another function:
+
+```rust
+use cosmwasm::mock::MockStorage;
+use cw_storage::{bucket, Bucket};
+
+fn people<'a, S: Storage>(storage: &'a mut S) -> Bucket<'a, S, Data> {
+    bucket(b"people", storage)
+}
+
+fn do_stuff() -> Result <()> {
+    let mut store = MockStorage::new();
+    people(&mut store).save(b"john", &Data{
+        name: "John",
+        age: 314,
+    })?;
+    OK(())
+}
+```
+
+### Singleton
```

Comment:
Hmm, singleton implies global (or static) state to avoid the need of carrying around an instance. Global state is dangerous if we cache instances. But this is not what happens for this class as far as I can see. It just creates an instance that keeps a private `key` field. So I think this thing needs a different name.

Reply:
It is global state, just on disk (not memory).

Is there another term for this in the DB? I am happy to change this now, as I am bumping to 0.2.0 with the other PR - time for more breaking changes :)

(This class is not part of this PR, but let's open an issue, handle it before a v0.2.0)